### DISCUSSION TITLE
SF5 and SF6 configuration in Sx1262

### DISCUSSION THREAD

**Discussion by vikasrohal** [Lora-net/LoRaMac-node#1229] [Q&A]
Hi all,
Sx1262 works on SF6 and SF5 as well, but there is no implementation of SFs in standard Firmware in any of country region.
Is there any other method to use SF5 and SF6 ?

Why SF 7 not possible on LoRa

if( dr == DR_7 )
{
modem = MODEM_FSK;
Radio.SetRxConfig( modem, 50000, phyDr * 1000, 0, 83333, 5, rxConfig->WindowTimeout, false, 0, true, 0, 0, false, rxConfig->RxContinuous );
}
else
{
modem = MODEM_LORA;
Radio.SetRxConfig( modem, rxConfig->Bandwidth, phyDr, 1, 0, 8, rxConfig->WindowTimeout, false, 0, false, 0, 0, true, rxConfig->RxContinuous );

    }

**Comment by Rgrndaal** [comment]
Bedankt voor uw email.

Op dit moment ben ik niet aanwezig op kantoor. Vanaf maandag 10 januari ben ik weer terug op kantoor. Gedurende deze periode heb ik slechts beperkt toegang tot mijn email.

-----------------------------

Thanks for your e-mail. 

I'm currently out of office and will return on Monday, January 10th. During this period I will have limited access to my e-mail.

Met vriendelijke groet,
Kind regards,

CLICKEY solutions

Remco Groenendaal
***@***.***

CLICKEY solutions Office: +31-162-511517
Lissenveld 47c, 4941 VL Raamsdonksveer, The Netherlands http://www.clickey.eu

**Comment by mluis1** [comment]
The SX126x radios do support LoRa spreading factors `SF5` and `SF6`. 
The radio drivers provided on this project allow the usage of `SF5` and `SF6`.
https://github.com/Lora-net/LoRaMac-node/blob/edff6ce161773e888687a9cd352ce2e2234e6b0d/src/radio/sx126x/sx126x.h#L347-L360

**However, LoRaWAN specifications do not use these spreading factors.**

The answer to your 2nd question is yes, spreading factor `SF7` is supported and used by LoRaWAN specifications.  
It looks like that your are mixing different things `DR_7` is not equal to `SF7`.   
The LoRaWAN specifications defines the data rate as an index which ranges from 0 up to 15. Each index corresponds to physical radio data rate (LoRa,  FSK or LR-FHSS).

Looking at LoRaWAN Regional Parameters Specification you will see for each region the mapping between the index and the real data rate.
Example for EU868 region:
![image](https://user-images.githubusercontent.com/5763295/148074835-0fe1d089-56ae-40dc-b6a5-d4183da5a5ef.png)

As you can observe `DR_7` corresponds to FSK data rate while `DR_5` corresponds to `SF7BW125` and `DR_6` to `SF7BW250`
